### PR TITLE
(Code) Change meta attribute => partial metadata

### PR DESCRIPTION
**Pull request by ekinanp**
Supports #684.

There's a few more changes that need to be made (see the issue). These changes only make sense after the code here's released.

### CODE REVIEW

File: cmd/internal/find/primary/meta.go [hunk 1]
```diff
@@ -70,58 +70,19 @@ const metaDetailedDescription = `
 NOTE: https://puppetlabs.github.io/wash/tutorials/02_find/meta-primary
 contains a hands-on tutorial of the meta primary.
 
-The meta primary constructs a predicate on the entry's metadata. By
```
Comment (ekinanp) [inline cmd/internal/find/primary/meta.go]: This was all deleted b/c much of this stuff will be in the meta primary tutorial (seems like it makes sense to put it there). Am open to other ideas of restructuring this.

Comment (MikaelSmith) [reply cmd/internal/find/primary/meta.go]: Ok, I agree most of that isn't needed as a reference, just the first time you encounter the concept.

File: plugin/entryBase.go [hunk 2]
```diff
@@ -55,61 +55,34 @@ func NewEntry(name string) EntryBase {
...
 		e.SetTTLOf(defaultOpCode(op), 15*time.Second)
 	}
 	return e
 }
 
+func (e *EntryBase) partialMetadata() JSONObject {
+	if e.specifiedPartialMetadata != nil {
+		return e.specifiedPartialMetadata
+	}
+	return e.attributes.ToMap()
```
Comment (MikaelSmith) [inline plugin/entryBase.go]: Why do we return attributes instead?

File: plugin/entryBase.go [hunk 3]
```diff
@@ -55,61 +55,34 @@ func NewEntry(name string) EntryBase {
...
+}
+
 // ENTRY INTERFACE
 
-// Metadata returns the entry's meta attribute (see plugin.EntryAttributes).
-// Override this if e has additional metadata.
+// Metadata returns the entry's partial metadata. Override this if e has
```
Comment (MikaelSmith) [inline plugin/entryBase.go]: It's worth calling out somewhere that it's normal to set partial metadata and override Metadata, and why that's the case.

**Review by MikaelSmith — CHANGES_REQUESTED** [review]

File: plugin/entryBase.go (hunk 2 above)
Comment (ekinanp) [reply plugin/entryBase.go]: That's the whole "attributes subset of metadata" thing. Edge-case is if someone specifies just the attributes, but they don't set partial metadata, that likely means the attributes are the metadata. I'll add a comment.

Comment (ekinanp) [reply plugin/entryBase.go]: Updated.

File: plugin/entryBase.go (hunk 3 above)
Comment (ekinanp) [reply plugin/entryBase.go]: Updated.

**Review by MikaelSmith — APPROVED** [review]

**Pull request merged**